shared/utils/response-v3.ts

```diff
@@ -28,10 +34,21 @@ export const isFieldResponseV3Equal = (
     case BasicField.Children:
       return _.isEqual(l.answer, r.answer)
     case BasicField.Attachment: {
-      const rAnswer = r.answer as typeof l.answer
+      const lAnswer = l.answer as ParsedClearAttachmentResponseV3['answer']
+      const rAnswer = r.answer as HydratedClearAttachmentResponseV3['answer']
+
+      const lMd5 = crypto
+        .createHash('md5')
+        .update(Buffer.from(lAnswer.content))
+        .digest()
+
+      const rMd5 = crypto
+        .createHash('md5')
+        .update(Buffer.from(rAnswer.content.data))
+        .digest()
+
```

Comment:
This part feels a bit strange in principle... because equality should be a symmetric function, L and R should have the same type. But in this case, we presume asymmetry  of L and R while it not being reflected in the argument types (both of them have argument types `FieldResponseV3`). This means any user of this function has to know which value to put on the left and which one to put on the right.

Should we make it the responsibility of the backend to re-transform it's input into a format amenable to this function prior to calling it? Then we leave the key part of the code allowing for symmetry `r.answer as typeof l.answer` alone. Then we can move the parsed and hydrated and clear attachment types to only be used in the backend.

Reply:
Moved to caller 247077854